src/EventStore.Core/ClusterVNodeStartup.cs

```diff
@@ -150,7 +123,13 @@ public class ClusterVNodeStartup<TStreamId> : IStartup, IHandle<SystemMessage.Sy
 							return Results.BadRequest("Redaction is only available via Unix Sockets");
 						return await next(c).ConfigureAwait(false);
 					});
+				});
+
+			// allow all subsystems to register their legacy controllers before finalizing the legacy pipeline
+			foreach (var subsystem in _subsystems)
+				subsystem.Configure(app);
 
+			app.UseEndpoints(ep => {
```

Comment:
i'd prefer not to have two useendpoints calls because its kind of confusing - the subsystems can probably go after userouting?